### ISSUE TITLE
Remove all uppercase text in buttons and links.

### ISSUE THREAD

**Issue by karmatosed** [Automattic/wp-calypso#4869]
Some buttons and links in Calypso are uppercase. This isn't great for those with any reading issues, some accessibility issues or even those whose language may not be the one the site loads in, due to location. It also could be interpreted as shouting at people, capital letters being a way to show shouting. That's not a great thing to portray for us.

I suggest we change but keep a universal way. I totally agree they should all be the same, I just feel we shouldn't have them all in uppercase.

I also found uppercasing in cases like 'REMOVE' used for people management. We should I suggest also not use uppercase there. We also have instances like 'ADD NEW ITEM'. Again this should I suggest be removed from being uppercase.

If people agree, I am happy about organising a patch to de-uppercase. I feel it would be something worth to do.

**Comment by karmatosed** [comment]
Adding examples:

<img width="780" alt="2016-04-19 at 19 53" src="https://cloud.githubusercontent.com/assets/253067/14651598/61b6d67e-0668-11e6-92c5-daa86c61efd6.png">

<img width="782" alt="2016-04-19 at 19 53" src="https://cloud.githubusercontent.com/assets/253067/14651610/7513060c-0668-11e6-80c3-1b46c2aa6233.png">

<img width="289" alt="2016-04-19 at 19 54" src="https://cloud.githubusercontent.com/assets/253067/14651628/8bdc717a-0668-11e6-8fd3-e12ce54acc4b.png">

**Comment by rickybanister** [comment]
This is worth some further discussion. Obviously the accessibility issues are important, but we'll need to take a careful look at rendering those buttons and notices and other things in lower case—the descenders throw off the visual alignment, especially in the presence of icons. 

Not to say that trumps the accessibility concerns, just that we'll have to take some time to do it right (things were made uppercase because it was easier to get them aligned in the buttons).

**Comment by karmatosed** [comment]
@rickybanister absolutely agree to consider it. I'm happy to even do a PR we can dry run and iterate on, if people are interested.

**Comment by drw158** [comment]
> things were made uppercase because it was easier to get them aligned in the buttons

I don't think this was the only reason (I didn't create the uppercase buttons). 

There are benefits to having uppercase buttons — they are distinguishable from regular text if used consistently. The best example is Material Design. All of their buttons are uppercase so that they are interpreted as buttons.

https://www.google.com/design/spec/components/buttons.html#buttons-button-types

There are several other examples of this, such as the buttons on iOS.

There are several 'is-borderless` button instances in Calypso, that would look like regular text, if the uppercase was removed.

![screen shot 2016-04-22 at 3 02 43 pm](https://cloud.githubusercontent.com/assets/618551/14753512/801edafa-089b-11e6-99dd-7491e877a402.png)

To address the accessibility concerns, maybe we can use a darker color, increase font weight, or increase font size.

**Comment by karmatosed** [comment]
@drw158 thanks for your input. You are right, all caps is an issue beyond just what we do. It is something a lot of other places do and have been called out over. Some change, some don't. I hope we can at least look at it more and I'm really pleased we are discussing this.

The designer in me totally gets wanting to have the clean state of uppercase, but I personally have issues reading capital letters like this. This makes me very sensitive to this, however I do not think we should just design for me though of course. I bring this up because it goes beyond my normal and is an issue for a lot of people. I also have concerns it creates 'Shouty UI' and I am not sure that's something we should create.

There's some interesting variation in thought on uppercase. Some even say the brain interprets it differently - super intriguing. But, no there isn't a great consensus in the major players. You are totally right. Also, well brains work differently so we have to find something that works for as many as possible - interesting challenge. 

The thing is, uppercase doesn't really work better for any brain or hasn't been proven to, but it has been proven to cause some people issues. When you consider it like that, to me it feels like we should evaluate using uppercase for actions.

I do not feel that the accessibility concerns are answered by a darker color, font weight or font size. They are still hard to make out the letters. The problem is it makes a lot of letters uniform or closer to each other in shape. Unfortunately, our font makes this even more of an issue and this actually can be made worse with font weight or size and not better with color.

Aside from the uppercase/lowercase all point, we actually have this super odd mix as shown by my links above. In talking to @rickybanister, he mentioned using lowercase a lot more lately. If we're moving that way then perhaps we should go for it. I am aware that in doing this we have to look at the cases you brought up where it would not stand out at all now. Perhaps though, if only being uppercase is the only thing making it stand out, that's not really the best anyway?

**Comment by rickybanister** [comment]
To give some specific examples of places where I have been seeing sentence case make more sense than upper case were the compact notices and the form toggle labels. 

There are compelling arguments for both cases when it comes to buttons (compact buttons, specifically) so I'm not really sure where I stand there. 

That said, I think we could work to remove the non-bold, non-button uppercase from ui components and approach this more slowly.

**Comment by drw158** [comment]
I agree with those 2 examples. We should be more selective about using uppercase.

**Comment by karmatosed** [comment]
So what would be the criteria for using uppercase? I'm keen to work out beyond a visual point of view, why uppercase is better for some cases and how we distinguish that. It probably isn't visual alone, but if it was I'd query why we consider that as a case to go against improving accessibility.

**Comment by drw158** [comment]
It's not just for aesthetics. The main reason, and perhaps the only reason, is to create text that is distinguishable, or to call out certain text.

This sums up my feelings on all caps: http://practicaltypography.com/all-caps.html

In the original pattern library for Calypso, all caps is used for cap­tions, head­ers, foot­ers, and other la­bels. I still think that is a good rule of thumb.

![screen shot 2016-04-25 at 11 04 00 am](https://cloud.githubusercontent.com/assets/618551/14790217/e80c9198-0ad5-11e6-876a-5eb7d80bb276.png)

**Comment by karmatosed** [comment]
For reference and to find out how deep this issue could go, I did a quick search and found 86 cases of 'text-transform: uppercase' in a lot of different instances. If worthwhile, I can audit further to see what each is used for.

**Comment by shaunandrews** [comment]
> It's not just for aesthetics. The main reason, and perhaps the only reason, is to create text that is distinguishable, or to call out certain text.

This is me telling you something in a normal voice. THIS IS ME TELLING YOU SOMETHING IN A FIRM, LOUD VOICE.

I think that distinction makes a ton of sense, especially with the REQUIRED example above. I'd love to see an audit and remove the uppercase text as needed.

**Comment by drw158** [comment]
I think we should look at this again. Discussion seems to have stopped. I'm 👍  for removing uppercase on all buttons. If someone could make a PR, we can get a feel for it. The only issue I see is borderless buttons without an icon will look just like regular text, but that might be ok in the context they are used in. Perhaps those borderless buttons should be blue or something like that.

**Comment by karmatosed** [comment]
I'm happy doing a PR, I want to get to know the codebase more for Calypso and this sounds like a great way to do that.

**Comment by lancewillett** [comment]
Addressed by https://github.com/Automattic/wp-calypso/pull/15023
